src/websocket/notification.gateway.ts

```diff
@@ -1,35 +1,60 @@
-import { Injectable } from '@nestjs/common';
-import {
-  ConnectedSocket,
-  SubscribeMessage,
-  WebSocketGateway,
-  WebSocketServer,
-} from '@nestjs/websockets';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
 import { Server, Socket } from 'socket.io';
+import { AuthService } from '../auth/auth.service';
+import { UserRepository } from '../user/user.repository';
 
 @Injectable()
 @WebSocketGateway()
 export class NotificationGateway {
```

Comment:
in my opinion, it is better to change the name of it to `WebSocketGateway` (and rename the entire file to `WebSocketGateway`)

```suggestion
export class WebSocketGateway {
```

Reply:
you should not connect WebSocket with only notifications 

it is better call it as  WebSocket events

Reply:
Maybe something like `WebSocketEventGateway()` is good enough?

Reply:
I think`WebSocketEventGateway()` will fit